src/Administration/Resources/app/administration/src/module/sw-cms/component/sw-cms-inherit-wrapper/index.ts

```diff
@@ -0,0 +1,144 @@
+                    config?: CmsSlotConfig;
+                }
+            }>,
+            required: true,
+        },
+        contentEntity: {
+            type: Object as PropType<{
+                slotConfig?: {
+                    [slotId: string]: CmsSlotConfig;
+                };
+            }>,
+            required: false,
+            default: null,
+        },
+        field: {
+            type: String,
+            required: true,
+        },
+        fieldPath: {
+            type: String,
+            default() {
+                return 'value';
+            },
+        },
+        label: {
+            type: String,
+            required: false,
+        },
+    },
+    data() {
+        return {
+            showModal: false,
+        };
+    },
+    computed: {
+        currentValue: {
+            get() {
+                return get(this.runtimeConfig, this.fullPath);
+            },
+            set(value: string) {
+                set(this.runtimeConfig, this.fullPath, value);
+
+                this.$emit(EVENTS.UPDATE, value);
+            },
+        },
+        baseConfig() {
+            return this.element.translated?.config;
+        },
+        childConfig() {
+            return this.contentEntity?.slotConfig?.[this.element.id];
+        },
+        runtimeConfig() {
+            return this.element.config;
+        },
+        supportsInheritance() {
+            return !!this.contentEntity;
+        },
+        canInheritField() {
+            return has(this.baseConfig, this.fullPath);
+        },
+        /**
+         * Fields are inherited if the layout is used on a content page (product, category, landing page)
+         * and the field is not overridden in the <entity>.slot_config
+         */
+        isInherited() {
+            return this.contentEntity && isUndefined(get(this.childConfig, this.fullPath));
+        },
+        fullPath() {
+            return this.field.concat('.', this.fieldPath);
+        },
+    },
+    methods: {
+        onInheritanceRestore() {
+            this.showModal = false;
+
+            set(this.runtimeConfig, this.fullPath, get(this.baseConfig, this.fullPath, null));
+            unset(this.childConfig, this.field);
+
+            if (isEmpty(this.childConfig)) {
+                unset(this.contentEntity, 'slotConfig');
+            }
+
+            this.$emit(EVENTS.RESTORE);
+        },
+        onInheritanceRemove() {
+            if (!this.contentEntity) {
+                return;
+            }
+
+            if (!this.contentEntity.slotConfig) {
+                this.contentEntity.slotConfig = {};
+            }
+
+            if (!this.contentEntity.slotConfig[this.element.id]) {
+                this.contentEntity.slotConfig[this.element.id] = {};
+            }
+
+            set(
+                this.childConfig!,
+                this.fullPath,
+                get(this.baseConfig, this.fullPath, {
+                    value: null,
```

Comment:
in line 111 you are using just null as fallback and here an object with the value null. Should it be more unified?

Reply:
In this case, the object `{ value: null }` is a default return value for the `get` function, which is reading the field from the parent config (`cms_slot_translation.config`). There may be cases where the `cms_slot_translation.config` is not holding all possible config fields - this way the UI will still be responsive to changes.

In line 111, the `fullPath` is including the `value` part, so the structure is actually the same 😄